Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -298,6 +299,117 @@ - (void)testUserChangePassword {
     }
 }
 
+/// A sync user should be able to successfully change their own password.
+- (void)testOtherUserChangePassword {
+    // Create admin user.
+    {
+        // Admin token user.
+        NSURL *adminTokenFileURL = [[RLMSyncTestCase rootRealmCocoaURL] URLByAppendingPathComponent:@"sync/admin_token.base64"];
+        NSString *adminToken = [NSString stringWithContentsOfURL:adminTokenFileURL encoding:NSUTF8StringEncoding error:nil];
+        XCTAssertNotNil(adminToken);
+        RLMSyncCredentials *credentials = [RLMSyncCredentials credentialsWithAccessToken:adminToken identity:@"test"];
+        XCTAssertNotNil(credentials);
+        RLMSyncUser *adminTokenUser = [self logInUserForCredentials:credentials server:[RLMObjectServerTests authServerURL]];
+
+        // Create admin/admin "normal" user. Won't have admin priviledges until further below.
+        RLMSyncCredentials *creds = [RLMSyncCredentials credentialsWithUsername:@"admin" password:@"admin" register:YES];
+        RLMSyncUser *adminUser = [self logInUserForCredentials:creds server:[RLMObjectServerTests authServerURL]];
+        XCTAssertFalse(adminUser.isAdmin);
+        [adminUser logOut];
+
+        // User should be created very quickly but not necessarily instantly.
+        [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:1.0]];
```

Comment:
That'd be pretty difficult AFAICT because I think the reason it's not instantaneous is that the transaction log has been successfully received but not necessarily applied, or the node.js runloop hasn't yet advanced to the latest state. There's no convenient way to hook into that from a client.